pkg/controllers/machinesetsync/machineset_sync_controller.go

```diff
@@ -259,9 +261,17 @@ func (r *MachineSetSyncReconciler) syncMachineSets(ctx context.Context, mapiMach
 }
 
 // reconcileMAPIMachineSetToCAPIMachineSet reconciles a MAPI MachineSet to a CAPI MachineSet.
+//
+//nolint:funlen
```

Comment:
Can we refactor anything in here (abstract out to helper funcs etc) to avoid this?

Reply:
Actually I think if you rebase this on top of the deletion PR and merge after that one has you'll avoid it as I've done some cleanup there.

Reply:
This is a bug fix so I'd like to keep it small. Also I'd like to backport this to 4.19, so I don't want to change things up too much.
I think we can do the refactoring in yours yes, but I don't think this should be blocked on that one.